backends/graphics/surfacesdl/surfacesdl-graphics.cpp

```diff
@@ -2522,23 +2525,33 @@ bool SurfaceSdlGraphicsManager::notifyEvent(const Common::Event &event) {
 			return true;
 		}
 
-		// Alt-S: Create a screenshot
+		// Alt-S: Create a screenshot - Win32
 		if (event.kbd.hasFlags(Common::KBD_ALT) && event.kbd.keycode == 's') {
-			char filename[20];
+			Common::String filename;
+
+#ifdef WIN32
+			Common::String screenshotsPath = OSystem_Win32().getScreenshotsPath();
```

Comment:
You want to access the unique OSystem instance here, from which the OSystem_SDL (and thus OSystem_Win32) derives. This instance is accessed by using g_system. Then you can cast it to OSystem_SDL to get access to the getScreenshotsPath() function.

And since you now have a virtual function there is no need to use OSystem_Win32 explicitly and you can use the same code on all systems.

(the same comment applies to openglsdl-graphics.cpp)